specs/derivation.md

```diff
@@ -473,37 +473,49 @@ updated, such that the batch-sender authentication is always accurate to the exa
-By default, the rollup operates on calldata retrieved from [batcher transactions][g-batcher-transaction] in the block,
-for each transaction:
+In the *L1 Retrieval* stage, we read the block we get from the outer stage (L1 traversal), and
+extract data from its [batcher transactions][g-batcher-transaction]. A batcher
+transaction is one with the following properties:
 
-- The receiver must be the configured batcher inbox address.
-- The sender must match the batcher address loaded from the system config matching the L1 block of the data.
+https://github.com/ethereum/execution-specs/blob/3fe6514f2d9d234e760d11af883a47c1263eff51/src/ethereum/frontier/fork_types.py#L52C31-L52C31
 
-Each data-transaction is versioned and contains a series of [channel frames][g-channel-frame] to be read by the
-Frame Queue, see [Batch Submission Wire Format][wire-format].
+- The
+  [`to`](https://github.com/ethereum/execution-specs/blob/3fe6514f2d9d234e760d11af883a47c1263eff51/src/ethereum/frontier/fork_types.py#L52C31-L52C31)
+  field is equal to the configured batcher inbox address.
 
-#### Eclipse: Blob Retrieval
+- The sender, as recovered from the transaction signature (`v`, `r`, and `s`), is the batcher
+  address loaded from the system config matching the L1 block of the data.
+
+Each batcher transaction is versioned and contains a series of [channel frames][g-channel-frame] to
+be read by the Frame Queue, see [Batch Submission Wire Format][wire-format]. Each batcher
+transaction in the block is processed by passing its calldata on to the next phase.
 
-With the Eclipse upgrade the retrieval stage is extended to support an additional DA source: [EIP-4844] blobs.
-Upon each traversed transaction in the batch inbox:
+#### Eclipse: Blob Retrieval
 
-- The receiver must be the configured batcher inbox address.
-- The sender must match the batcher address loaded from the system config matching the L1 block of the data.
-- Calldata is utilized, even if contained in a blob tx, and passed on to the next stage before any blob-data is.
-- If the transaction-type is `0x03` (`BLOB_TX_TYPE`): retrieve the blobs that match the `blob_versioned_hashes`.
-  - Blobs may be retrieved from different sources. Retrieval from a local beacon-node,
-    through the `/eth/v1/beacon/blob_sidecars/` endpoint, with `indices` filter to skip unrelated blobs, is recommended.
-  - The rollup node SHOULD cryptographically verify if the retrieved blobs match the versioned hashes.
+With the Eclipse upgrade the retrieval stage is extended to support an additional DA source:
+[EIP-4844] blobs. With this extension, the above batcher transaction processing is modified to
+consider only batcher transactions with transaction-type != `0x03` (`BLOB_TX_TYPE`). Only after
+fully processing these non-blob transaction do we then consider blob transactions, each of which is
+processed as follows:
```

Comment:
Preserving transaction ordering would be a good property, to avoid uncertainty during L1 block-building of what the effects on L2 inputs are.

Reply:
ack. updated the spec here.  Will update impl next.